src/test/java/com/team766/hal/wpilib/WpilibJoystickTest.java

```diff
@@ -0,0 +1,27 @@
+package com.team766.hal.wpilib;
+
+import com.team766.hal.JoystickAbstractTest;
+import edu.wpi.first.wpilibj.simulation.JoystickSim;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+
+// TODO: AdvantageKit breaks this test. Try re-enabling this after upgrade to 2025.
+@Disabled
+public class WpilibJoystickTest extends JoystickAbstractTest {
+    private JoystickSim driver;
+
+    @BeforeEach
+    public void setUp() {
+        joystick = new Joystick(0);
+        driver = new JoystickSim(0);
+    }
+
+    @Override
+    protected void setAxis(int axis, double value) {
+        driver.setRawAxis(axis, value);
+        while (joystick.getAxis(axis) != value) {
+            updateDriverStationData();
```

Comment:
WPILib's simulation model of DriverStation is double-buffered. You set all the values that you want using JoystickSim, DriverStationSim, etc, then you flush the new values to the robot code using `DriverStationSim.notifyNewData();`.

This is currently dead code (this test case is `@Disabled`) because AdvantageKit sorta breaks unit tests that depend on DriverStation inputs: AdvantageKit makes the update of the DriverStation buffer asynchronous. The `while` loop was my attempt to workaround their issues, but I still couldn't get it working.

https://github.com/Mechanical-Advantage/AdvantageKit/blob/v3.2.0/docs/COMMON-ISSUES.md#unit-testing

From reading their release notes, it looks like AdvantageKit has switched to a synchronous update model for their 2025/4.0 version, but it hasn't been released yet (it's still in beta). I'm going to revisit this test after we upgrade (see the `TODO` comment)